cli/src/main/java/com/box/l10n/mojito/cli/command/checks/CliCheckerExecutor.java

```diff
@@ -0,0 +1,83 @@
+package com.box.l10n.mojito.cli.command.checks;
+
+import com.box.l10n.mojito.cli.command.CommandException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class CliCheckerExecutor {
+
+    static Logger logger = LoggerFactory.getLogger(CliCheckerExecutor.class);
+
+    private final List<CliChecker> cliCheckerList;
+
+    private final ExecutorService executorService;
+
+    private String notificationText;
+
+    public CliCheckerExecutor(List<CliChecker> cliCheckerList, int numOfThreads) {
+        this.cliCheckerList = cliCheckerList;
+        this.executorService = Executors.newFixedThreadPool(numOfThreads);
+    }
+
+    public List<String> executeChecks() {
+        try {
+            List<Future<CliCheckResult>> futures = startChecks(executorService);
+            return analyzeResults(retrieveCheckResults(futures));
+        } catch (InterruptedException | ExecutionException e) {
+            throw new CommandException("Failed to retrieve results from check executor: " + e.getMessage(), e);
+        } finally {
+            executorService.shutdown();
```

Comment:
what happens if this is called twice? 

else spring a bunch of facilities to run async task that may help hide some of this setup (executor service, etc)

Reply:
I'll move the creation of the thread pool to be inside the method, in that case a threadpool is created and shutdown on each call to executeChecks.